rpc/backend/blocks.go

```diff
@@ -296,28 +298,82 @@ func (b *Backend) EthMsgsFromTendermintBlock(
 		tx, err := b.clientCtx.TxConfig.TxDecoder()(tx)
 		if err != nil {
 			b.logger.Debug("failed to decode transaction in block", "height", block.Height, "error", err.Error())
+			// try to check if there is synthetic eth tx in tx result
+			ethMsg, additional := b.parseSyntheticTxFromBlockResults(txResults, i, tx, block)
```

Comment:
Just for my understanding, is it okay to use tx here even if `err != nil` . Also maybe we should call this txDecoded to prevent shadowing?